src/Internal/Wallet/Cip30.purs

```diff
@@ -149,28 +146,30 @@ hexStringToAddress =
   ((addressFromBytes <<< rawBytesAsCborBytes) <=< hexToRawBytes)
 
 -- | Get collateral using CIP-30 `getCollateral` method.
--- | Throws on `Promise` rejection by wallet, returns `Nothing` if no collateral
--- | is available.
+-- | Throws on `Promise` rejection by wallet or conversion failure.
+-- | Returns `Nothing` if no collateral is available.
 getCollateral :: Cip30Connection -> Aff (Maybe (Array TransactionUnspentOutput))
 getCollateral conn = do
-  mbUtxoStrs <- toAffE $ getCip30Collateral conn
-  let
-    (mbUtxoBytes :: Maybe (Array RawBytes)) =
-      join $ map (traverse hexToRawBytes) mbUtxoStrs
-  -- de-serialize UTxOs
-  liftEffect $ for mbUtxoBytes \collateralUtxos -> do
-    for collateralUtxos \bytes -> do
-      maybe (throw "Unable to convert UTxO") pure =<<
-        Deserialization.UnspentOuput.convertUnspentOutput
-          <$> fromBytesEffect (unwrap bytes)
+  mbUtxoStrs <- toAffE $ _getCollateral maybeFfiHelper conn
+  liftEffect $ (for mbUtxoStrs $ traverse txOutputFromHexString)
+
+getCollateralYoroi
+  :: Cip30Connection -> Aff (Maybe (Array TransactionUnspentOutput))
+getCollateralYoroi conn = do
+  mbUtxoStrs <- toAffE $ _getCollateralYoroi maybeFfiHelper conn
+  liftEffect $ (for mbUtxoStrs $ traverse txOutputFromHexString)
+
+-- | Converts a hexadecimal string to a `TransactionUnspentOutput`,
+-- | lifting any conversion failures into an error in the `Effect` monad.
+txOutputFromHexString :: String -> Effect TransactionUnspentOutput
```

Comment:
Should not this be in `Deserialization` module or something like that?

Reply:
Or is this too specific to be usable by another modules?

Reply:
Cool deduplication by the way!

Reply:
@uhbif19 Deserialization is for CSL domain -> CTL domain conversion. This is Hex String -> CTL.